swent-epfl-groupe07/eduverse-android
# Add Publication Deletion Feature

This PR implements the ability for users to delete their own publications from their profile, with proper UI feedback and error handling.

## Changes

### User Interface
- Added delete button in publication detail dialog (only visible to publication owners)
- Implemented confirmation dialog to prevent accidental deletions
- Added error message display for deletion failures
- Integrated with existing deletion functionality from backend

### Testing
- Added comprehensive test suite covering:
 - Delete button visibility logic
 - Confirmation dialog behavior
 - Success/failure states
 - Dialog dismissal handling
 - Error message display

## Implementation Details

The deletion flow:
1. User clicks delete button in publication detail
2. Confirmation dialog appears with warning
3. If confirmed:
  - Backend deletion process starts
  - UI updates based on operation result
  - On success: dialog closes and profile refreshes
  - On failure: error message shown to user

## Screenshots

<div style="display: flex; gap: 10px;">
 <img src="https://github.com/user-attachments/assets/4a85f19a-0c2f-4816-91b3-cac2eb3a188e" width="200" />
 <img src="https://github.com/user-attachments/assets/c72d970e-a68e-47d9-b918-f4fb41757981" width="200" />
 <img src="https://github.com/user-attachments/assets/27618ff5-311e-4d00-a538-257502ef57ae" width="200" />
 <img src="https://github.com/user-attachments/assets/060519e5-5539-4c59-9a54-9e10b02251ab" width="200" />
</div>

## Testing Instructions

1. Open your profile
2. Select any publication you own
3. Try to delete it with the new delete button
4. Verify confirmation dialog appears
5. Test both confirmation and cancellation
6. Verify proper error handling

## Related Issues

Closes #232

## Checklist

- [x] UI implementation complete
- [x] Code formatted with ktfmt
